mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergFilterFactory.java

```diff
@@ -0,0 +1,241 @@
+        .startAnd()
+        .equals("salary", PredicateLeaf.Type.LONG, 3000L)
+        .equals("salary", PredicateLeaf.Type.LONG, 4000L)
+        .end().build();
+
+    And expected = (And) Expressions
+        .and(Expressions.equal("salary", 3000L), Expressions.equal("salary", 4000L));
+    And actual = (And) HiveIcebergFilterFactory.generateFilterExpression(arg);
+
+    assertEquals(actual.op(), expected.op());
+    assertEquals(actual.left().op(), expected.left().op());
+    assertEquals(actual.right().op(), expected.right().op());
+  }
+
+  @Test
+  public void testOrOperand() {
+    SearchArgument.Builder builder = SearchArgumentFactory.newBuilder();
+    SearchArgument arg = builder
+        .startOr()
+        .equals("salary", PredicateLeaf.Type.LONG, 3000L)
+        .equals("salary", PredicateLeaf.Type.LONG, 4000L)
+        .end().build();
+
+    Or expected = (Or) Expressions
+        .or(Expressions.equal("salary", 3000L), Expressions.equal("salary", 4000L));
+    Or actual = (Or) HiveIcebergFilterFactory.generateFilterExpression(arg);
+
+    assertEquals(actual.op(), expected.op());
+    assertEquals(actual.left().op(), expected.left().op());
+    assertEquals(actual.right().op(), expected.right().op());
+  }
+
+  @Test
+  public void testStringType() {
+    SearchArgument.Builder builder = SearchArgumentFactory.newBuilder();
+    SearchArgument arg = builder.startAnd().equals("string", PredicateLeaf.Type.STRING, "Joe").end().build();
+
+    UnboundPredicate expected = Expressions.equal("string", "Joe");
+    UnboundPredicate actual = (UnboundPredicate) HiveIcebergFilterFactory.generateFilterExpression(arg);
+
+    assertPredicatesMatch(expected, actual);
+  }
+
+  @Test
+  public void testFloatType() {
+    SearchArgument.Builder builder = SearchArgumentFactory.newBuilder();
+    SearchArgument arg = builder.startAnd().equals("float", PredicateLeaf.Type.FLOAT, 1200D).end().build();
+
+    UnboundPredicate expected = Expressions.equal("float", 1200D);
+    UnboundPredicate actual = (UnboundPredicate) HiveIcebergFilterFactory.generateFilterExpression(arg);
+
+    assertPredicatesMatch(expected, actual);
+  }
+
+  @Test
+  public void testBooleanType() {
+    SearchArgument.Builder builder = SearchArgumentFactory.newBuilder();
+    SearchArgument arg = builder.startAnd().equals("boolean", PredicateLeaf.Type.BOOLEAN, true).end().build();
+
+    UnboundPredicate expected = Expressions.equal("boolean", true);
+    UnboundPredicate actual = (UnboundPredicate) HiveIcebergFilterFactory.generateFilterExpression(arg);
+
+    assertPredicatesMatch(expected, actual);
+  }
+
+  /*@Test
+  public void testDateType() {
+    SearchArgument.Builder builder = SearchArgumentFactory.newBuilder();
+    SearchArgument arg = builder.startAnd().equals("date", PredicateLeaf.Type.DATE,
+            Date.valueOf("2015-11-12")).end().build();
+
+    UnboundPredicate expected = Expressions.equal("date", LocalDate.of(2015,11,12));
```

Comment:
@rdblue @rdsr I added tests for the `Date` and `Timestamp` types but when these are run I get errors like: 

```
java.lang.IllegalArgumentException: Cannot create expression literal from java.time.LocalDate: 2015-11-12
	at org.apache.iceberg.expressions.Literals.from(Literals.java:83)
	at org.apache.iceberg.expressions.UnboundPredicate.<init>(UnboundPredicate.java:39)
	at org.apache.iceberg.expressions.Expressions.equal(Expressions.java:159)
        at org.apache.iceberg.mr.hive.TestHiveIcebergFilterFactory.testDateType(TestHiveIcebergFilterFactory.java:211)
```

I noticed [here](https://github.com/apache/iceberg/blob/c20927801a369104e5ea510470e1cf7c8e28b808/api/src/test/java/org/apache/iceberg/expressions/TestLiteralSerialization.java#L38) in another test that Date's etc.  are actually passed as Strings - is that the correct option to be using in this case?

Reply:
Yes, it is a good idea to use a string instead of passing a `LocalDate`. The intent was to avoid tying the API to date/time representations from a specific library.